CorsixTH/Lua/app.lua

```diff
@@ -396,6 +398,28 @@ function App:initUserLevelDir()
   return true
 end
 
+--! Tries to initialize the user and built in campaign directories, returns true on
+--! success and false on failure.
+function App:initUserCampaignDir()
+  local conf_path = self.command_line["config-file"] or "config.txt"
+  self.user_campaign_dir = self.config.campaigns or
+      conf_path:match("^(.-)[^" .. pathsep .. "]*$") .. "Campaigns"
+
+  if self.user_campaign_dir:sub(-1, -1) == pathsep then
+    self.user_campaign_dir = self.user_campaign_dir:sub(1, -2)
+  end
+  if lfs.attributes(self.user_campaign_dir, "mode") ~= "directory" then
+    if not lfs.mkdir(self.user_campaign_dir) then
+      print("Notice: Campaign directory does not exist and could not be created.")
+      return false
+    end
+  end
+  if self.user_campaign_dir:sub(-1, -1) ~= pathsep then
+    self.user_campaign_dir = self.user_campaign_dir .. pathsep
+  end
+  return true
+end
```

Comment:
Don't copy/paste code!!

Refactor `App:initUserLevelDir` to a common part and a userLevel-specific part, and use the former also here.

Reply:
Tbh yes that was me trying to just get it working ages ago. Then neglected to actually tidy it. I'll work on that.